contrib/zsh-completion/_terraforming

```diff
@@ -0,0 +1,93 @@
+#compdef terraforming
+
+function _terraforming() {
+  local context curcontext=$curcontext state line
+  typeset -A opt_args
+  local ret=1
+
+  _arguments -C \
+    '1: :__terraforming_sub_commands' \
+    '*:: :->args' \
+    && ret=0
+
+  case $state in
+    (args)
+      case $words[1] in
+        (asg|dbpg|dbsg|dbsn|ec2|ecc|ecsn|eip|elb|iamg|iamgm|iamgp|iamip|iamp|iamr|iamrp|iamu|iamup|igw|nacl|nif|r53r|r53z|rds|rs|rt|rta|s3|sg|sn|vpc)
+          _arguments -C \
+            '--merge=[(TFSTATE) tfstate file to merge]:tfstate file to merge:_files' \
+            '--overwrite[Overwrite existing tfstate]' \
+            '--no-overwrite[Do not overwrite existing tfstate]' \
+            '--tfstate[Generate tfstate]' \
+            '--no-tfstate[Do not generate tfstate]' \
+            '--profile[(PROFILE) AWS credentials profile]:AWS credentials profile:__profiles' \
+            && ret=0
+        ;;
+        (help)
+          _arguments -C \
+            '1: :__terraforming_sub_commands' \
+            '(-)*:: :->null_state' \
+            && ret=0
+        ;;
+      esac
+      ;;
+  esac
+
+  return ret
+}
+
+__terraforming_sub_commands() {
+  local -a _c
+
+  _c=(
+    'asg:AutoScaling Group'
+    'dbpg:Database Parameter Group'
+    'dbsg:Database Security Group'
+    'dbsn:Database Subnet Group'
+    'ec2:EC2'
+    'ecc:ElastiCache Cluster'
+    'ecsn:ElastiCache Subnet Group'
+    'eip:EIP'
+    'elb:ELB'
+    'help:Describe available commands or one specific command'
+    'iamg:IAM Group'
+    'iamgm:IAM Group Membership'
+    'iamgp:IAM Group Policy'
+    'iamip:IAM Instance Profile'
+    'iamp:IAM Policy'
+    'iamr:IAM Role'
+    'iamrp:IAM Role Policy'
+    'iamu:IAM User'
+    'iamup:IAM User Policy'
+    'igw:Internet Gateway'
+    'nacl:Network ACL'
+    'nif:Network Interface'
+    'r53r:Route53 Record'
+    'r53z:Route53 Hosted Zone'
+    'rds:RDS'
+    'rs:Redshift'
+    'rt:Route Table'
+    'rta:Route Table Association'
+    's3:S3'
+    'sg:Security Group'
+    'sn:Subnet'
+    'vpc:VPC'
+  )
```

Comment:
This might be better to fetch resource list from `terraforming` command dynamically :bulb: 

```bash
_c=("${(@f)$(terraforming --help | grep '  terraforming' | sed -E 's/^  terraforming //g' | sed -E 's/ *(\[COMMAND\])? *# /:/g')}")
```